src/components/pages/block/txs/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { l2PublicClient } from "@/lib/chains";
+import Link from "next/link";
+import {
+  Table,
+  TableBody,
+  TableCaption,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { formatTimestamp } from "@/lib/utils";
+import { Card } from "@/components/ui/card";
+
+const BlckTxs = async ({ number }: { number: bigint }) => {
+  async function getGasUsed(hash: any, gasPrice: bigint) { 
```

Comment:
You should await all tx receipts using `Promise.all`:

```
const receipts = await Promise.all(
  block.transactions.map(({ hash }) =>
    l2PublicClient.getTransactionReceipt({ hash }),
  ),
);
```

Then using the receipts and the block data, build a transactions array to feed the Table.